markets/perps-market/contracts/modules/AsyncOrderSettlementPythModule.sol

```diff
@@ -103,41 +108,74 @@ contract AsyncOrderSettlementPythModule is
+        asyncOrder.reset();
+    }
+
+    /// @dev Updates the position and market data
+    function _processPositionUpdate(
+        uint256 price,
+        SettleOrderRuntime memory runtime,
+        PerpsMarket.Data storage market,
+        uint256 originalMarketSize
+    ) internal returns (MarketUpdate.Data memory) {
+        // Update position data
+        MarketUpdate.Data memory updateData = market.updatePositionData(
             runtime.accountId,
             runtime.newPosition
         );
-        perpsAccount.updateOpenPositions(runtime.marketId, runtime.newPosition.size);
 
+        // Calculate the market size delta (change in market size)
+        int256 marketSizeDelta = market.size.toInt() - originalMarketSize.toInt();
+
+        // Emit MarketUpdated event
         emit MarketUpdated(
-            runtime.updateData.marketId,
+            updateData.marketId,
             price,
-            runtime.updateData.skew,
-            runtime.updateData.size,
-            runtime.sizeDelta,
-            runtime.updateData.currentFundingRate,
-            runtime.updateData.currentFundingVelocity,
-            runtime.updateData.interestRate
+            updateData.skew,
+            market.size,
+            marketSizeDelta,
+            updateData.currentFundingRate,
+            updateData.currentFundingVelocity,
+            updateData.interestRate
         );
 
-        runtime.settlementReward = AsyncOrder.settlementRewardCost(settlementStrategy);
+        return updateData;
+    }
 
+    /// @dev Processes the order fees and settlement rewards
+    function _processFees(
+        SettleOrderRuntime memory runtime,
+        AsyncOrder.Data storage asyncOrder,
+        PerpsMarketFactory.Data storage factory
+    ) internal {
         // if settlement reward is non-zero, pay keeper
         if (runtime.settlementReward > 0) {
             factory.withdrawMarketUsd(ERC2771Context._msgSender(), runtime.settlementReward);
         }
 
-        {
-            // order fees are total fees minus settlement reward
-            uint256 orderFees = runtime.totalFees - runtime.settlementReward;
-            GlobalPerpsMarketConfiguration.Data storage s = GlobalPerpsMarketConfiguration.load();
-            s.collectFees(orderFees, asyncOrder.request.referrer, factory);
-        }
-
-        // trader can now commit a new order
-        asyncOrder.reset();
+        // order fees are total fees minus settlement reward
+        uint256 orderFees = runtime.totalFees - runtime.settlementReward;
+        GlobalPerpsMarketConfiguration.Data storage s = GlobalPerpsMarketConfiguration.load();
+        s.collectFees(orderFees, asyncOrder.request.referrer, factory);
+    }
 
-        /// @dev two events emitted to avoid stack limitations
+    /// @dev Emit settlement events in a helper function to reduce stack depth
+    function _emitSettlementEvents(
+        SettleOrderRuntime memory runtime,
+        AsyncOrder.Data storage asyncOrder
```

Comment:
this can be marked as memory too. would be interesting to see gas comparison bc most if not all of these slots are warm.